### PR TITLE
Add phone page to register journey

### PR DESCRIPTION
**Pull request by wilpow-github**
### Context

As an overseas qualified teacher
I want to sign in/register a Teaching services account
So that I can apply for QTS

### Changes proposed in this pull request

Add a new page at /sign-in/register/phone matching the designs in the prototype

### Checklist

-   [x] Attach to Trello card
-   [ ] Rebased master
-   [ ] Cleaned commit history
-   [ ] Tested by running locally
-   [ ] Reminder created to manually clean any removed app settings post deployment

### CODE REVIEW

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Models/Mappings/SmsConfirmationPinMapping.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace TeacherIdentity.AuthServer.Models.Mappings;
+
+public class SmsConfirmationPinMapping : IEntityTypeConfiguration<SmsConfirmationPin>
+{
+    public void Configure(EntityTypeBuilder<SmsConfirmationPin> builder)
+    {
+        builder.ToTable("sms_confirmation_pins");
+        builder.HasKey(p => p.SmsConfirmationPinId);
+        builder.Property(p => p.MobileNumber).HasMaxLength(100).IsRequired();
```
Comment (wilpow-github) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Models/Mappings/SmsConfirmationPinMapping.cs:12]: 100 characters should be enough. Most of the validation happens in notification service, so I don't want to be too strict

**Comment by wilpow-github** [comment]
Sorry about the large number of files!

We're currently only storing the phone number on authentication state. Have not added a new column for User yet

File: dotnet-authserver/src/TeacherIdentity.AuthServer/AuthenticationState.cs [hunk 2]
```diff
@@ -102,6 +102,9 @@ public class AuthenticationState
     public string? StatedTrn { get; private set; }
     [JsonInclude]
     public HasPreviousNameOption? HasPreviousName { get; private set; }
+    public string? MobileNumber { get; private set; }
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/AuthenticationState.cs:105]: Think we need `[JsonInclude]` on this property.

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/EmailConfirmation.cshtml.cs [hunk 3]
```diff
@@ -37,11 +37,11 @@ public async Task<IActionResult> OnPost()
             return this.PageWithErrors();
         }
 
-        var verifyPinFailedReasons = await EmailVerificationService.VerifyPin(Email!, Code!);
+        var VerifyEmailPinFailedReasons = await UserVerificationService.VerifyEmailPin(Email!, Code!);
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/EmailConfirmation.cshtml.cs:40]: Nit
```suggestion
        var verifyEmailPinFailedReasons = await UserVerificationService.VerifyEmailPin(Email!, Code!);
```

File: dotnet-authserver/tests/TeacherIdentity.AuthServer.EndToEndTests/HostFixture.cs [hunk 4]
```diff
@@ -42,7 +42,7 @@ public class HostFixture : IAsyncLifetime
 
     public Mock<IDqtApiClient> DqtApiClient { get; } = new();
 
-    public IEmailVerificationService? EmailConfirmationService { get; private set; }
+    public IUserVerificationService? UserConfirmationService { get; private set; }
```
Comment (gunndabad) [inline dotnet-authserver/tests/TeacherIdentity.AuthServer.EndToEndTests/HostFixture.cs:45]:
```suggestion
    public IUserVerificationService? UserVerificationService { get; private set; }
```

**Review by gunndabad — APPROVED** [review]

**Pull request merged**